### PR TITLE
Improve sofb

### PR DESCRIPTION
**Pull request by fernandohds564**
New features:
 - check if any BPM being used in closed loop correction is not connected. Open loop in case this is True;
 - check if orbit distortion is larger than a given threshold (200um, max, by default). If it is, open loop;
 - select singular values by minimum value instead of quantity;
 - add Tikhonov regularization;
 - implements a PID controller for Closed Loop operation.

Note: Testing and debugging are still needed before merging.

### CODE REVIEW

File: siriuspy/siriuspy/sofb/csdev.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ class ETypes(_csdev.ETypes):
     """Local enumerate types."""
 
     ENBL_RF = _csdev.ETypes.OFF_ON
+    OPEN_CLOSED = ('OPEN', 'CLOSED')
```
Comment (anacso17) [inline siriuspy/siriuspy/sofb/csdev.py:19]: Shouldn't it be `OPEN_CLOSED = ('Open', 'Closed')`?

File: siriuspy/siriuspy/sofb/main.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ def __init__(
         super().__init__(acc, prefix=prefix, callback=callback)
         _log.info('Starting SOFB...')
         self._orbit = self._correctors = self._matrix = None
-        self._loop_state = self._csorb.ClosedLoop.Off
+        self._loop_state = self._csorb.LoopState.Off
```
Comment (anacso17) [inline siriuspy/siriuspy/sofb/main.py:30]: Isn't it `self._csorb.LoopState.Open`?

File: siriuspy/siriuspy/sofb/csdev.py (hunk 1 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/sofb/csdev.py:19]: putz, verdade!

File: siriuspy/siriuspy/sofb/main.py (hunk 2 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/sofb/main.py]: done! Thanks!

File: siriuspy/siriuspy/sofb/csdev.py (hunk 1 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/sofb/csdev.py]: done! Thanks!

Comment (xresende) [reply siriuspy/siriuspy/sofb/csdev.py]: > putz, verdade!

what do you mean?

**Pull request merged**